src/hooks/use-i18n-data.tsx

```diff
@@ -21,8 +21,23 @@ export const I18nDataProvider = ( { children }: { children: React.ReactNode } )
 	const [ locale, setLocale ] = useState< SupportedLocale >( 'en' );
 
 	const initI18n = useCallback( async ( localeKey: SupportedLocale ) => {
-		const newI18n = createI18n( getLocaleData( localeKey )?.messages );
+		const translations = getLocaleData( localeKey )?.messages;
+		const newI18n = createI18n( translations );
 		setI18n( newI18n );
+		// Update default I18n data to reflect language change when using
+		// I18n functions from `@wordpress/i18n` package.
+		// Note we need to update this in both the renderer and main processes.
+		if ( translations ) {
+			defaultI18n.setLocaleData( translations );
+			getIpcApi().setDefaultLocaleData( translations );
+		} else {
+			// In case we don't find translations, we reset the locale data to
+			// fallback to the default translations.
+			defaultI18n.resetLocaleData();
```

Comment:
Should we remove that, as we do it via ipc?

Reply:
https://github.com/Automattic/studio/pull/504/files#diff-4897e47ea3fb6cdbb1798e8dce993fa538a1d6d56b425eabbf82d945889ec508R726

Reply:
As far as I've tested we need both. The object `defaultI18n` object is created globally so we need to update it for each process, the main and renderer.